### ISSUE TITLE
SSL error while downloading chromium for the first time

### ISSUE THREAD

**Issue by ravisumit33** [pyppeteer/pyppeteer#105]
While downloading chromium for the first time, i got the following error:

`OpenSSL.SSL.Error: [('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')]`

I had to use [https://github.com/kiwi0fruit/pyppdf/blob/11d082f7a35cdac2ae3e7ffa7022c1d1e9747cd2/pyppdf/patch_pyppeteer/patch_pyppeteer.py#L59](https://github.com/kiwi0fruit/pyppdf/blob/11d082f7a35cdac2ae3e7ffa7022c1d1e9747cd2/pyppdf/patch_pyppeteer/patch_pyppeteer.py#L59) to solve my issue.

As seen in the above link, It uses HTTPS to download while pyppeteer uses HTTP. Can't HTTPS be used for pyppeteer to solve this issue?

**Comment by Mattwmaster58** [comment]
Yes, HTTPS can be used. This has been fixed on the pup2.1.1 branch.

**Comment by reformedot** [comment]
`urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='storage.googleapis.com', port=443): Max retries exceeded with url: /chromium-browser-snapshots/Mac/588429/chrome-mac.zip (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])")))`

When I run puppeteer for the first time this exception is executed.

How I can manage to fix it? The link provided by @ravisumit33 it's no longer available.

Thanks!

**Comment by jessrosenfield** [comment]
Is this the same issue as #70? I get this error 
```
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='storage.googleapis.com', port=443): Max retries exceeded with url: /chromium-browser-snapshots/Linux_x64/588429/chrome-linux.zip (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])")))
```
on w/ the following Python version (AWS Lambda Python3.7)
```
3.7.7 (default, May 6 2020, 12:32:55)
[GCC 4.8.3 20140911 (Red Hat 4.8.3-9)]
```
with
```
pyppeteer==0.2.2
urllib3==1.25.9
```
I didn't, however, get this issue on macOS.

Was this bug fixed by #76? 
I'm considering monkey-patching my own `download_zip` that uses `certifi` into my use of Pyppeteer 0.2.2 (a-la https://github.com/kiwi0fruit/pyppdf/blob/master/pyppdf/patch_pyppeteer/patch_pyppeteer.py). Alternatively, I can install a development branch with a fix if you think any of them are stable enough. I'd love to hear your thoughts? Also, do you have a guesstimate as to when the next release of Pyppeteer will be?

**Comment by Mattwmaster58** [comment]
It's a different error it would seem than #70, but likely related. `certifi` is used in the `pup2.1.1` branch for downloading Chromium. You could try installing from the `pup2.1.1` branch here for a huge update, some parts are pretty polished and others are uncharted territory. Case in point, I don't think the auto downloading logic is in there. But you should still be able to run one of the console scripts in the package to download chromium. No support for this ATM though.

> Also, do you have a guesstimate as to when the next release of Pyppeteer will be?

Depends when/if contributors have time. Likely >2 months

**Comment by jessrosenfield** [comment]
Got it. Thanks so much for such a speedy reply!

**Comment by FredrikNoren** [comment]
Running this before using pyppeteer solved it for me:

```
PYPPETEER_DOWNLOAD_HOST=http://storage.googleapis.com pyppeteer-install
```
